src/page/GroupEdit.tsx

```diff
@@ -144,9 +141,38 @@ const GroupEdit = () => {
     setSelectedTags(tags || []);
     setIsPublicGroup(!password);
     setSelectedImageFile(null);
-    navigate(`/group/${id}/admin`); // 초기화 후 라우터로 이동
+    navigate(`/group/${id}/admin`);
   };
 
+  useEffect(() => {
+    if (id && !isLoading) {
+      const {
+        title: groupTitle,
+        imageSrc,
+        capacity,
+        tags,
+        introduce,
+        information,
+        password,
+      } = data.data;
+
+      const initialData = {
+        title: groupTitle || '',
+        num: capacity ? capacity.toString() : '0',
+        secretKey: password || '',
+        intro: introduce || '',
+        group: information || '',
+        previewImage: imageSrc || null,
+        selectedTags: tags || '',
+      };
+
+      setInputs(initialData);
+      setSelectedTags(initialData.selectedTags);
+      setIsPublicGroup(!password);
+      setPreviewImage(initialData.previewImage);
+    }
+  }, [data]);
+
```

Comment:
P1: 
1. 우선 id 값이 없는 경우는 위에서 얼리 리턴 처리를 해주었기 때문에, id가 없는 경우에는 useEffect까지 도달하지 않아요. 
그래서 !id는 불필요한 조건인 것 같아요.

2. data를 의존성에 추가해주셔서 data가 들어온 경우 useEffect가 실행될텐데, 해당 함수 내부에서 다시 한 번 동일한 데이터 값을 불러오는 이유가 무엇인지 궁금해요! (동일한 맥락으로, 이 과정에서 다시 initialData를 정의하는 이유도 궁금합니다.)

++ 그룹 정보를 수정하는 뷰이기 때문에, 저장버튼을 눌러 수정 뷰를 벗어나는 경우와 에러가 발생해서 해당 뷰에 머무르는 두 가지 경우가 있을 것 같아요. 그렇다면 수정 뷰에 머무르는 동안은 서버에서 받아오는 정보의 내용이 달라질 일이 없지 않을까요!?

3. 새로고침했을 때 제대로 들어오지 않던 데이터가 제대로 들어온 이유는 useEffect때문에 맞을 거에요 ! 그렇지만, 앞서 남겨둔 이유들을 고려해보았을 때 useEffect 내부에서 다시 한 번 그룹 정보를 불러오는 것은 옳지 않다고 판단하여 아래와 같이 수정해보았어요.
```suggestion
  useEffect(() => {
    if (!isGetDetailLoading) {
      setInputs({
        title: originTitle,
        num: capacity,
        secretKey: password,
        intro: introduce,
        group: information,
        previewImage: imageSrc,
        selectedTags: tags,
      });
      setIsPublicGroup(isPublicRoom);
      setPreviewImage(imageSrc);
      setSelectedTags(tags);
    }
  }, [isGetDetailLoading]);
```

Reply:
1. 넵 
++2. useEffect 내부에서 다시 데이터를 추출하고 initialData를 정의했던 이유는 새로고침 시 데이터를 초기화하는 문제가 있었기 때문에 이것저것 코드에 적용시키다보니 불필요한 코드가 추가돼서 들어갔던 것 같습니다! 아름이 코드 참고하면서 수정해 보겠습니다!

감사합니다👍